rxlifecycle-android-lifecycle/src/test/java/com/trello/rxlifecycle2/android/lifecycle/RxLifecycleObserverFragmentTest.java

```diff
@@ -0,0 +1,128 @@
+
+        controller.start();
+        controller.resume();
+        controller.pause();
+        controller.stop();
+        controller.destroy();
+
+        testObserver.assertValues(
+                Lifecycle.Event.ON_CREATE,
+                Lifecycle.Event.ON_START,
+                Lifecycle.Event.ON_RESUME,
+                Lifecycle.Event.ON_PAUSE,
+                Lifecycle.Event.ON_STOP,
+                Lifecycle.Event.ON_DESTROY
+        );
+    }
+
+    // Tests bindUntil for any given RxLifecycleObserver Fragment implementation
+    private void testBindUntilEvent(LifecycleOwner owner) {
+        Fragment fragment = (Fragment) owner;
+        ActivityController<?> controller = startFragment(fragment);
+
+        TestObserver<Object> testObserver = observable.compose(new RxLifecycleObserver(owner).bindUntilEvent(Lifecycle.Event.ON_STOP)).test();
+
+        testObserver.assertNotComplete();
+        controller.start();
+        testObserver.assertNotComplete();
+        controller.resume();
+        testObserver.assertNotComplete();
+        controller.pause();
+        testObserver.assertNotComplete();
+        controller.stop();
+        testObserver.assertComplete();
+    }
+
+    // Tests bindToLifecycle for any given RxLifecycle Fragment implementation
+    private void testBindToLifecycle(LifecycleOwner owner) {
+        Fragment fragment = (Fragment) owner;
+        LifecycleProvider<Lifecycle.Event> provider = new RxLifecycleObserver(owner);
+        ActivityController<?> controller = startFragment(fragment);
+
+        TestObserver<Object> createObserver = observable.compose(provider.bindToLifecycle()).test();
+
+        controller.start();
+        createObserver.assertNotComplete();
+        TestObserver<Object> startObserver = observable.compose(provider.bindToLifecycle()).test();
+
+        controller.resume();
+        createObserver.assertNotComplete();
+        startObserver.assertNotComplete();
+        TestObserver<Object> resumeObserver = observable.compose(provider.bindToLifecycle()).test();
+
+        controller.pause();
+        createObserver.assertNotComplete();
+        startObserver.assertNotComplete();
+        resumeObserver.assertComplete();
+        TestObserver<Object> pauseObserver = observable.compose(provider.bindToLifecycle()).test();
+
+        controller.stop();
+        createObserver.assertNotComplete();
+        startObserver.assertComplete();
+        pauseObserver.assertComplete();
+        TestObserver<Object> stopObserver = observable.compose(provider.bindToLifecycle()).test();
+
+        controller.destroy();
+        createObserver.assertComplete();
+        stopObserver.assertComplete();
+    }
+
+    // Easier than making everyone create their own shadows
+    private ActivityController<FragmentActivity> startFragment(Fragment fragment) {
+        ActivityController<FragmentActivity> controller = Robolectric.buildActivity(FragmentActivity.class);
```

Comment:
Honestly, I hate Robolectric as a result of this project. I'm *this* close to just making them run on an emulator.